test/plugin/test_in_tail.rb

```diff
@@ -487,39 +487,42 @@ def test_z_refresh_watchers
       @loop = Coolio::Loop.new
     end
 
-    flexstub(Time) do |timeclass|
-      timeclass.should_receive(:now).with_no_args.and_return(Time.new(2010, 1, 2, 3, 4, 5), Time.new(2010, 1, 2, 3, 4, 6), Time.new(2010, 1, 2, 3, 4, 7))
+    stub(Time) do |timeclass|
+      timeclass.now.returns(Time.new(2010, 1, 2, 3, 4, 5), Time.new(2010, 1, 2, 3, 4, 6), Time.new(2010, 1, 2, 3, 4, 7))
 
-      flexstub(Fluent::NewTailInput::TailWatcher) do |watcherclass|
+      stub(Fluent::NewTailInput::TailWatcher) do |watcherclass|
```

Comment:
Your usage is wrong. Use the following patch:

```diff
diff --git a/test/plugin/test_in_tail.rb b/test/plugin/test_in_tail.rb
index d9b6086..80dceca 100644
--- a/test/plugin/test_in_tail.rb
+++ b/test/plugin/test_in_tail.rb
@@ -488,19 +488,33 @@ class TailInputTest < Test::Unit::TestCase
     end
 
     stub(Time) do |timeclass|
-      timeclass.now.returns(Time.new(2010, 1, 2, 3, 4, 5), Time.new(2010, 1, 2, 3, 4, 6), Time.new(2010, 1, 2, 3, 4, 7))
+      nows = [
+        Time.new(2010, 1, 2, 3, 4, 5),
+        Time.new(2010, 1, 2, 3, 4, 6),
+        Time.new(2010, 1, 2, 3, 4, 7),
+      ]
+      timeclass.now do
+        nows.shift
+      end
 
       stub(Fluent::NewTailInput::TailWatcher) do |watcherclass|
         EX_PATHS.each do |path|
-          file_position_entry = stub(Fluent::NewTailInput::FilePositionEntry)
-          watcherclass.new(path, EX_RORATE_WAIT, file_position_entry, anything, true, 1000, anything, anything).once do
+          new = watcherclass.new(path,
+                                 EX_RORATE_WAIT,
+                                 is_a(Fluent::TailInput::FilePositionEntry),
+                                 anything,
+                                 true,
+                                 1000,
+                                 anything,
+                                 anything) do
             mock('TailWatcher') do |watcher|
               watcher.attach(anything).once
-              watcher.__send__(:unwatched=).at_least(0)
+              watcher.__send__(:unwatched=, true).at_least(0)
               watcher.line_buffer.at_least(0)
               watcher.close.at_least(0)
             end
           end
+          new.once
         end
         plugin.refresh_watchers
       end
@@ -510,14 +524,21 @@ class TailInputTest < Test::Unit::TestCase
       end
 
       stub(Fluent::NewTailInput::TailWatcher) do |watcherclass|
-        file_position_entry = stub(Fluent::NewTailInput::FilePositionEntry)
-        watcherclass.new('test/plugin/data/2010/01/20100102-030406.log', EX_RORATE_WAIT, file_position_entry, anything, true, 1000, anything, anything, anything).once do
+        new = watcherclass.new('test/plugin/data/2010/01/20100102-030406.log',
+                               EX_RORATE_WAIT,
+                               is_a(Fluent::TailInput::FilePositionEntry),
+                               anything,
+                               true,
+                               1000,
+                               anything,
+                               anything) do
           mock('TailWatcher') do |watcher|
             watcher.attach(anything).once
-            watcher.__send__(:unwatched=).at_least(0)
+            watcher.__send__(:unwatched=, true).at_least(0)
             watcher.line_buffer.at_least(0)
           end
         end
+        new.once
         plugin.refresh_watchers
       end
 
```

  1. `timeclass.now.returns(now1, now2, now3)` doesn't define a stub that returns `now1`, then `now2` and then `now3`. The stub always returns `now1`.
  2. `stub(Fluent::NewTailInput::FilePositionEntry)` doesn't mean that any `FilePositionEntry` instances. Use `is_a(Fluent::TailInput::FilePositionEntry)`.
  3. `watcherclass.new.once do ... end` is wrong. `do ... end` is a block for `once` not `new`.
  4. `watcher.__send__(:unwatched=).at_least(0)` misses an expected argument of `#unwatched=`.
  5. The number of arguments of the second `watcherclass.new` is wrong. It passes 9 arguments but `new` is called with 8 arguments.

By the way, you should use `mock` instead of `stub` to check expected behavior. `stub` is used just for stub. It means that `stub` just provides features for running a test. It doesn't check anything. `mock` is used for checking behaviors. Search "stub mock difference" on Web search engine for details.